runtime/vm/classallocation.c

```diff
@@ -519,6 +519,16 @@ freeClassLoader(J9ClassLoader *classLoader, J9JavaVM *javaVM, J9VMThread *vmThre
 		classLoader->classRelationshipsHashTable = NULL;
 	}
 
+	/* Free the outliving loader set. */
+	if (NULL != classLoader->outlivingLoaders) {
+		if ((J9CLASSLOADER_OUTLIVING_LOADERS_PERMANENT != classLoader->outlivingLoaders)
+			&& J9_ARE_NO_BITS_SET((UDATA)classLoader->outlivingLoaders, J9CLASSLOADER_OUTLIVING_LOADERS_SINGLE_TAG)
+		) {
+			hashTableFree((J9HashTable *)classLoader->outlivingLoaders);
+		}
+		classLoader->outlivingLoaders = NULL;
+	}
```

Comment:
I think this should go after line 312, after the call to `freeSharedCacheCLEntries()`, just like in `jvmfree.c`.